state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs

```diff
@@ -133,24 +142,40 @@ impl MockAptosDataClient {
         }
     }
 
+    /// Clones the mock data client without timeout verification
+    fn clone_without_timeout_verification(&self) -> Self {
+        // Clone the mock data client and skip timeout verification
+        let mut aptos_data_client = self.clone();
+        aptos_data_client.skip_timeout_verification = true;
+
+        aptos_data_client
+    }
+
+    /// Emulates network latencies by sleeping for 10 -> 50 ms
     fn emulate_network_latencies(&self) {
-        if self.skip_emulate_network_latencies {
-            return;
+        if !self.skip_emulate_network_latencies {
+            thread::sleep(Duration::from_millis(create_range_random_u64(10, 50)));
```

Comment:
could this be tokio sleep? presumable other tasks could run while waiting for network

Reply:
Yeah, makes sense. 😄 I've made them `async` functions, updated them to use `tokio::time::sleep` and also updated the `emulate_network_request_timeout` method, too.